network/p2p/builder/libp2pNodeBuilder.go

```diff
@@ -463,13 +466,28 @@ func DefaultNodeBuilder(
 		}
 		builder.SetSubscriptionFilter(subscription.NewRoleBasedFilter(r, idProvider))
 
-		if dhtSystemActivation == DhtSystemEnabled {
-			builder.SetRoutingSystem(
-				func(ctx context.Context, host host.Host) (routing.Routing, error) {
-					return dht.NewDHT(ctx, host, protocols.FlowDHTProtocolID(sporkId), logger, metricsCfg.Metrics, dht.AsServer())
-				})
-		}
+		builder.configureRoutingSystem(r, dhtSystemActivation)
 	}
 
 	return builder, nil
 }
+
+func (b *LibP2PNodeBuilder) configureRoutingSystem(
+	role flow.Role,
+	dhtSystemActivation DhtSystemActivation,
+) {
+	if role != flow.RoleAccess && role != flow.RoleExecution {
+		return // routing only required for Access and Execution nodes
+	}
+
+	if dhtSystemActivation == DhtSystemEnabled {
+		b.SetRoutingSystem(func(ctx context.Context, host host.Host) (routing.Routing, error) {
+			return dht.NewDHT(ctx, host, protocols.FlowDHTProtocolID(b.sporkId), b.logger, b.metricsConfig.Metrics, dht.AsServer())
+		})
+	}
+
+	// bitswap requires a content routing system. this returns a stub instead of a full DHT
+	b.SetRoutingSystem(func(ctx context.Context, host host.Host) (routing.Routing, error) {
+		return none.ConstructNilRouting(ctx, host, nil, nil)
+	})
```

Comment:
is this supposed to be in an else block?

Reply:
[d732359](https://github.com/onflow/flow-go/pull/5827/commits/d732359ecb883310869610dd64f2c15019d20234)